### PR TITLE
Implement CI testing via Github Actions

### PR DESCRIPTION
**Pull request by RareBeeph**

### CODE REVIEW

File: .github/workflows/test-action.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        with: 
+          ref: 'actions-implementation'
+          
+      - uses: actions/setup-go@v3
+        with:
+          go-version-file: './go.mod'
+          
+      - run: go test -v ./cmd
```
Comment (aricodes-oss) [inline .github/workflows/test-action.yml]: Just for future proofing, the command `go test ./...` will run all tests from the current package and sub packages. That way this won't need to be updated for future test suites.

File: .github/workflows/test-action.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: Test Action
+on: push
+
+jobs: 
+  test-job:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with: 
+          ref: 'actions-implementation'
```
Comment (aricodes-oss) [inline .github/workflows/test-action.yml]: Nope. That prevents this from running on any other branch than this one, remove that qualifier. GA is pretty smart and knows what you mean when you ask it to checkout the repo

**Review by aricodes-oss — CHANGES_REQUESTED** [review]
One major issue, one minor issue, and one nitpick - name your PRs better. A good PR title is a good commit message, and is what your commits will be squashed under when we merge them.

File: cmd/root_test.go [hunk 3]
```diff
@@ -46,6 +46,8 @@ func trimWhitespace(s []byte) []byte {
 }
 
 func TestFlagArgs(t *testing.T) {
+	var testExitCode int
```
Comment (ariataylor96) [inline cmd/root_test.go]: The "dead man's switch" concept here is a clever pattern for solving tricky control flow problems, but is a bit too manual. Your test suite code should be treated as a library being run by the testing framework - in this case, the `T` context object you receive from the framework has a method called `Fail()` that you can use to mark a test as failed, removing the need for this mechanism.

**Review by ariataylor96 — CHANGES_REQUESTED** [review]

**Review by ariataylor96 — APPROVED** [review]

**Pull request merged**